modules/js-and-css/speculation-rules/helper.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Helper functions used for Speculation Rules.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Prints the speculation rules in a cross-browser compatible way.
+ *
+ * For browsers that do not support speculation rules yet, the rules will not be loaded.
+ *
+ * @since n.e.x.t
+ *
+ * @return array Associative array of speculation rules by type.
+ */
+function plsr_get_speculation_rules() {
+	$base_href_exclude_paths = array(
+		'/wp-login.php',
+		'/wp-admin/*',
```

Comment:
Right, this is a tricky one though. Nonces can also be sent through any other kind of parameter. I'm not sure how far we should go here by default, as what you're suggesting would just be a "best guess", but not actually reliable. So potentially it's better to support just what WordPress core does and require plugins to take care of this themselves.

I realize this may not work either (as adoption is hard), but also questioning whether it's reasonable to start "patching" for what plugins may do here.

Reply:
It's true that nonces can be sent via any parameter, but the `wp_nonce_url()` function uses the `_wpnonce` parameter by default. So I think it makes sense to omit them by default.